lib/ace/ext/code_lens.js

```diff
@@ -155,30 +153,44 @@ function attachToEditor(editor) {
         session.widgetManager = new LineWidgets(session);
         session.widgetManager.attach(editor);
     }
+    session.codeLensProviders = [];
     editor.renderer.on("afterRender", renderWidgets);
-    editor.$codeLenseClickHandler = function(e) {
-        var command = e.target.lenseCommand;
+    editor.$codeLensClickHandler = function(e) {
+        var command = e.target.lensCommand;
         if (command)
             editor.execCommand(command.id, command.arguments);
     };
-    editor.container.addEventListener("click", editor.$codeLenseClickHandler);
+    editor.container.addEventListener("click", editor.$codeLensClickHandler);
     editor.$updateLenses = function() {
         var session = editor.session;
         if (!session) return;
-        var codeLensProvider = session.codeLensProvider || session.$mode.codeLensProvider;
-        var lenses = codeLensProvider && codeLensProvider.provideCodeLenses(session);
-        
-        var cursor = session.selection.cursor;
-        var oldRow = session.documentToScreenRow(cursor);
-        exports.setLenses(session, lenses);
-        
-        var lastDelta = session.$undoManager && session.$undoManager.$lastDelta;
-        if (lastDelta && lastDelta.action == "remove" && lastDelta.lines.length > 1)
-            return;
-        var row = session.documentToScreenRow(cursor);
-        var lineHeight = editor.renderer.layerConfig.lineHeight;
-        var top = session.getScrollTop() + (row - oldRow) * lineHeight;
-        session.setScrollTop(top);
+        var providersToWaitNum = session.codeLensProviders.length;
+        var lenses = [];
+        session.codeLensProviders.forEach(function(provider) {
+            provider.provideCodeLenses(session, function(currentLenses) {
+                currentLenses.forEach(function(lens) {
+                    lenses.push(lens);
+                });
+                providersToWaitNum--;
+                if (providersToWaitNum == 0) {
```

Comment:
I'm a little surprised by the `==` instead of `===`, but I see it's allowed from the eslint rules: https://github.com/ajaxorg/ace/blob/master/.eslintrc#L32

Those rules are very different from the ones we use in Cloud9: https://code.amazon.com/packages/AWSCloud9Public/blobs/mainline/--/packages/eslint-config/index.js

Is there are reason for this difference?